src/main/java/cn/nukkit/entity/mob/EntityPiglin.java

```diff
@@ -41,4 +44,14 @@ public float getHeight() {
     public String getName() {
         return "Piglin";
     }
+
+    @Override
+    public boolean isBaby() {
+        return this.getDataFlag(DATA_FLAGS, Entity.DATA_FLAG_BABY);
+    }
+
+    @Override
+    public boolean isPreventingSleep(Player player) {
+        return !this.isBaby()/*TODO: Should this check player's golden armor?*/;
+    }
```

Comment:
If the player can't sleep without golden armor when he is near a Piglin, then yes.